packages/mdc-dialog/foundation.ts

```diff
@@ -210,45 +184,15 @@ class MDCDialogFoundation extends MDCFoundation {
     this.detectScrollableContent_();
   }
 
-  /** @private */
-  detectStackedButtons_() {
-    // Remove the class first to let us measure the buttons' natural positions.
-    this.adapter_.removeClass(cssClasses.STACKED);
-
-    const areButtonsStacked = this.adapter_.areButtonsStacked();
-
-    if (areButtonsStacked) {
-      this.adapter_.addClass(cssClasses.STACKED);
-    }
-
-    if (areButtonsStacked !== this.areButtonsStacked_) {
-      this.adapter_.reverseButtons();
-      this.areButtonsStacked_ = areButtonsStacked;
-    }
-  }
-
-  /** @private */
-  detectScrollableContent_() {
-    // Remove the class first to let us measure the natural height of the content.
-    this.adapter_.removeClass(cssClasses.SCROLLABLE);
-    if (this.adapter_.isContentScrollable()) {
-      this.adapter_.addClass(cssClasses.SCROLLABLE);
-    }
-  }
-
-  /**
-   * @param {!Event} evt
-   * @private
-   */
-  handleInteraction(evt) {
+  handleInteraction(evt: MouseEvent | KeyboardEvent) {
     const isClick = evt.type === 'click';
-    const isEnter = evt.key === 'Enter' || evt.keyCode === 13;
+    const isEnter = (evt as KeyboardEvent).key === 'Enter' || (evt as KeyboardEvent).keyCode === 13;
```

Comment:
Extract these checks to separate functions to make the code more readable. Ditto below.

Reply:
Done

Reply:
Ken wasn't wild about extracting these out into `util` functions, so I reverted that change.